code/modules/power/supermatter/supermatter.dm

```diff
@@ -360,7 +364,8 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
 		air_update_turf()
 
 	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
-		if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
+		var/obj/item/organ/eyes/eyes = l.getorganslot("eye_sight")
+		if(!istype(l.glasses, /obj/item/clothing/glasses/meson) && !istype(eyes, /obj/item/organ/eyes/robotic))
```

Comment:
Like with the previous check, this should really check for the `status` of the organ, not its path. The path doesn't guarantee anything.